packages/components/docs/jest.md

```diff
@@ -0,0 +1,105 @@
+    1. Using snapshot tests for large nested components can results in very large snapshot files which are hard to
+        review when it comes time to change something or update that snapshot
+    1. Don't forget to commit the related `.snap` files for your test cases. Without these, TeamCity will have trouble
+        verifying that your tests are valid.
+    1. Treat the `.snap` files as code when it comes to changes made to existing snapshots and newly created test cases.
+        This means that you should review the new `.snap` files when a new case is added to make sure the contents exist
+        and are as expected. This also means that changes to these files should be reviewed during code review of a
+        pull request.
+    1. Don't manually update `.snap` files. Review the content of the failure for an existing test and then use the
+        `yarn test -u` option to update the files. Note that if you are running the test case manually from IntelliJ,
+        there is an "update snapshot" link you can click that will use this option for you.
+    1. If you are getting local test failures and unexpected changes to a `.snap` file that seems unrelated to your set
+        of changes in your feature branch, be sure to locally run the `yarn install` command from the
+        `/packages/components` directory. There have been a few cases where an update to a package dependency can result
+        in some DOM changes, and without having those package updates installed locally, you can get conflicting results.
+1. You can run individual jest test cases from IntelliJ directly. When run in debug mode, you can set breakpoints in
+    the jest test code but also in the React component code or functions.
+    1. When doing enzyme testing, using `wrapper.debug()` can be used to show HTML for debugging purposes.
+    1. Note that under certain circumstances, like a series of snapshots tests, there may be a case where a test will
+        pass when run individually but fail when run with the rest of the related test cases. For these reason it is
+        good to run the full `yarn test` command on the `packages/components` directory before pushing your changes.
+
+## Mocks
+We have several examples of tests using `xhr-mock` for reading in realistic data that can be captured from the
+LabKey server to use in the various test cases. Having a jest spec file call `initUnitTestMocks()` in its setup
+(i.e. `beforeAll()`) will use the same mock data that `storybook` uses for mocking up responses to various API calls
+made during a component lifecycle.
+
+Note that several of our current tests run without mocking up all of the functions / APIs that they call. When this
+happens, it is still possible for a test to run and pass, but you will likely see outputs in the test run that look
+like the following:
+```
+Cannot log after tests are done. Did you forget to wait for something async in your test?
+```
```

Comment:
Perhaps add a "This is fine." message here if that's the verdict.  I would really like this to be "If this happens, you can do XYZ to avoid these messages",  but I'm not sure there is an easy, universal XYZ (Is it just setting up `xhr-mock` and proxying all requests perhaps?  I've not tried.)